Tasks/AzureAppServiceManageV0/azureappservicemanage.ts

```diff
@@ -202,4 +216,84 @@ async function run() {
     }
 }
 
+async function handleActionWithPublishProfileEndpoint(publishProfileEndpoint: AzureEndpoint, action: string, extensionList: string, extensionOutputVariables: string) {
+    let scmCreds: scmCredentials = await getSCMCredentialsFromPublishProfile(publishProfileEndpoint.PublishProfile);
+    let kuduService = new Kudu(scmCreds.uri, scmCreds.username, scmCreds.password);
+    let kuduServiceUtils = new KuduServiceUtils(kuduService);
+    let taskResult = true;
+    let errorMessage: string = "";
+
+    try {
+        switch(action) {
+            case "Start all continuous webjobs": {
+                await kuduServiceUtils.startContinuousWebJobs();
+                break;
+            }
+            case "Stop all continuous webjobs": {
+                await kuduServiceUtils.stopContinuousWebJobs();
+                break;
+            }
+            case "Install Extensions": {
+                var extensionOutputVariablesArray = (extensionOutputVariables) ? extensionOutputVariables.split(',') : [];
+                await kuduServiceUtils.installSiteExtensions(extensionList.split(','), extensionOutputVariablesArray);
+                break;
+            }
+            default: {
+                throw Error(tl.loc('InvalidActionForPublishProfileEndpoint'));
+            }
+        }
+    }
+     catch(exception) {
+        taskResult = false;
+        errorMessage = exception;
+    }
+
+    tl.debug('Completed action');
+    try {
+        switch(action) {
+            case "Install Extensions": {
+                if(kuduServiceUtils) {
+                    await kuduServiceUtils.updateDeploymentStatus(taskResult, null, { "type" : action });
+                }
+                break;
+            }
+            default: {
+                tl.debug(`deployment status not updated for action: ${action}`);
+            }
+        }
+    }
+    catch(error) {
+        tl.debug(error);
+    }
+    finally {
+        dispose();
+    }
+
+    if (!taskResult) {
+        tl.setResult(tl.TaskResult.Failed, errorMessage);
+    }
+}
+
+async function getSCMCredentialsFromPublishProfile(pubxmlFile: string): Promise<scmCredentials> {
+    let res;
+    await parseString(pubxmlFile, (error, result) => {
+        if(!!error) {
+            throw new Error("Failed XML parsing " + error);
+        }
+        res = result.publishData.publishProfile[0].$;
```

Comment:
hardcoded path can throw exception if any of them do not exist. You can wrap the logic to find `credentials` in try-catch and throw user friendly error